src/fluid/braginskii/bragLimiter.hpp

```diff
@@ -0,0 +1,49 @@
+// ***********************************************************************************
+// Idefix MHD astrophysical code
+// Copyright(C) Geoffroy R. J. Lesur <geoffroy.lesur@univ-grenoble-alpes.fr>
+// and other code contributors
+// Licensed under CeCILL 2.1 License, see COPYING for more information
+// ***********************************************************************************
+#ifndef FLUID_BRAGINSKII_BRAGLIMITER_HPP_
+#define FLUID_BRAGINSKII_BRAGLIMITER_HPP_
+
+#include "idefix.hpp"
+
+template<const Limiter limiter = Limiter::VanLeer>
+class BragLimiter {
+ public:
+  BragLimiter()=default;
+
+  KOKKOS_FORCEINLINE_FUNCTION static real MinModLim(const real dvp, const real dvm) {
+    real dq= 0.0;
+    // MinMod
+    if(dvp*dvm >0.0) {
+      real dq = ( fabs(dvp) < fabs(dvm) ? dvp : dvm);
+    }
+    return(dq);
+  }
+
+  KOKKOS_FORCEINLINE_FUNCTION static real VanLeerLim(const real dvp, const real dvm) {
+    real dq= 0.0;
+    dq = (dvp*dvm > ZERO_F ? TWO_F*dvp*dvm/(dvp + dvm) : ZERO_F);
+    return(dq);
+  }
+
+  KOKKOS_FORCEINLINE_FUNCTION static real McLim(const real dvp, const real dvm) {
+    real dq = 0;
+    if(dvp*dvm >0.0) {
+      real dqc = 0.5*(dvp+dvm);
+      real d2q = 2.0*( fabs(dvp) < fabs(dvm) ? dvp : dvm);
+      dq= fabs(d2q) < fabs(dqc) ? d2q : dqc;
+    }
+    return(dq);
+  }
+
+  KOKKOS_FORCEINLINE_FUNCTION static real Lim(const real dvp, const real dvm) {
+    if constexpr(limiter == Limiter::VanLeer) return(VanLeerLim(dvp,dvm));
+    if constexpr(limiter == Limiter::McLim) return(McLim(dvp,dvm));
+    if constexpr(limiter == Limiter::MinMod) return(MinModLim(dvp,dvm));
```

Comment:
Yes, it's copied, and it's the reason why I've been messing around with the #includes (see your above 2 comments). 

For the record, the problem is that the inclusion of the class declaration in slopeLimiter.hpp, leads to the inclusion of shockFlattening.hpp that itself depends on fluid.hpp, that depends on braginskiixxx.hpp. So if we do that, we get a circular dependency loop. I've been trying to avoid this loop by delaying the #include after the class declaration, but this is not a success with clang.

The key element is that in principle slope limiters should not need shockFlattening, except for the Riemann solvers, but it implies a severe restructuration of the slopeLimiter class, and of all the Riemann solvers. I therefore chose the "easy" way and copied only the slope limiter functions in braginskii. A proper way would be to refactor the Riemann solvers. Not today...

Reply:
fair enough, thanks !